src/components/Table/GridTableApi.ts

```diff
@@ -28,8 +28,8 @@ export function useGridTableApi<R extends Kinded>(): GridTableApi<R> {
 
 /** Provides an imperative API for an application page to interact with the table. */
 export type GridTableApi<R extends Kinded> = {
-  /** Scrolls row `index` into view; only supported with `as=virtual` and after a `useEffect`. */
-  scrollToIndex(index: number): void;
+  /** Scrolls row `index` into view; Defaults "smooth" behavior; only supported with `as=virtual` and after a `useEffect`. */
+  scrollToIndex(index: number | FlatIndexLocationWithAlign): void;
```

Comment:
Exposed all the options virtuoso offers since they seemed pretty standard.

Reply:
Historically we've avoided exposing 3rd party types to callers, b/c it leads to them getting coupled to ideally-internal dependencies that make it harder to change/swap out over the long-term.

Also, ideally Beam is making opinionated / "just works" defaults that pages that "want to scroll" get w/o having to tweak/ask for different behavior.

Is there something specific in `FlatIndex...` that pages need / "should be allowed" to override/pass? Fwiw kinda hoping not yet, but maybe?

If so, I'd like to pull those out into a Beam-specific type.

Reply:
Ah thats what looked off about it. Updated to use a custom type.

Specifically the only two that I think make sense are `behavior` and `position`.